src/odemis/driver/tescan.py

```diff
@@ -432,45 +454,39 @@ def _single_acquisition(self, channel):
                         (self._roi != (l, t, r, b)) or
                         (self._dt != dt) or
                         (self.pre_res != res)):
-                        self._device.ScStopScan()
-                        # flush remaining data in data buffer
-                        self.flush()
-                        # need to reset
-                        self._device.connection.socket_c.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
-                        self._device.connection.socket_d.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
                         self._device.ScScanLine(1, scaled_shape[0], scaled_shape[1],
                                              l + 1, t + 1, r + 1, b + 1, (dt / TESCAN_PXL_LIMIT), TESCAN_PXL_LIMIT, 0)
                         self._scaled_shape = scaled_shape
                         self._roi = (l, t, r, b)
                         self._dt = dt
                 else:
-                    if self.pre_res == (1, 1):
-                        self._device.ScStopScan()
-                        # flush remaining data in data buffer
-                        self.flush()
                     self._device.ScScanXY(0, scaled_shape[0], scaled_shape[1],
                                          l, t, r, b, 1, dt)
                 # we must stop the scanning even after single scan
                 # fetch the image (blocking operation), ndarray is returned
                 if res == (1, 1):
-                    sem_pxs = self._device.FetchArray(channel, TESCAN_PXL_LIMIT)
+                    sem_pxs = self._device.FetchImage(channel, TESCAN_PXL_LIMIT)
                     # Since we acquired TESCAN_PXL_LIMIT integrations of
                     # dt/TESCAN_PXL_LIMIT we now get the mean signal and return
                     # it as the result
+                    sem_pxs = numpy.frombuffer(sem_pxs, dtype=">u2")
                     sem_img = numpy.array([sem_pxs.mean()])
+                    logging.debug("Acquiring chamber image of %s", sem_img.shape)
```

Comment:
This log and the one just a few lines below needs to be polished. Here you could log the value directly.
In the log below, show the length instead of shape.